src/trainer.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This software is licensed under the terms of the MIT License.
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2011-2019, Lukas Weber <laochailan@web.de>.
+ * Copyright (c) 2012-2019, Andrei Alexeyev <akari@taisei-project.org>.
+ */
+
+#ifndef IGUARD_trainer_h
+#define IGUARD_trainer_h
+
+#include "taisei.h"
+
+typedef struct Trainer Trainer;
+
+struct Trainer {
+	int trainer_lives_stage;
+	int trainer_bombs_stage;
+	int trainer_hits_stage;
+
+	int trainer_lives_total;
+	int trainer_bombs_total;
+	int trainer_hits_total;
+};
```

Comment:
I would recommend something like that instead:

```suggestion
struct Trainer {
	struct {
		int lives;
		int bombs;
		int hits;
	} total, stage;
};
```

Keeping total and stage stats in separate sub-structs like this allows easily resetting them separately with a memset call:
```c
memset(&trailer.stage, 0, sizeof(trainer.stage));
```

The `trainer_` prefix on all fields is also unnecessary.